code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -99,44 +129,73 @@
 
 	var/break_doors = CANT //If CAN, they can attempt to open doors. If CANPLUS, they break the door down entirely
 	var/health_cap = 250 //Maximum possible health it can have. Because screw having a 1000 health mob
-/*	wanted_objects = list(
-		/obj/machinery/light,        // Bust out lights
+	var/busy = FALSE //Stop spamming the damn doorsmash
+	wanted_objects = list(
+		/obj/machinery/light,
+		/obj/machinery/door        // Bust out lights
 	)
 	search_objects = 1
 
 /mob/living/simple_animal/hostile/necro/zombie/CanAttack(var/atom/the_target)
+	if(the_target == creator)
+		return 0
+	if(ismob(the_target))
+		var/mob/living/M = the_target
+		if(ishuman(the_target)) //Checking for food
+			var/mob/living/carbon/human/H = the_target
+			if(H.isDead())
+				if(check_edibility(H))
+					if(isjusthuman(H)) //Always chow down on humans
```

Comment:
Because check_edibility checks if a regular zombie is 'full' just above (At max health, and max maxHealth)